cohorts/load.py

```diff
@@ -372,24 +709,33 @@ def plot_survival(self, on, col=None, col_equals=None, how="os", threshold=None)
             Whether to plot OS (overall survival) or PFS (progression free survival)
         threshold : int or 'median', optional
             Threshold of `col` on which to split the cohort
-
         """
         assert how in ["os", "pfs"], "Invalid choice of survival plot type %s" % how
-        plot_col, df = self.plot_init(on, col, col_equals)
+        plot_col, df = self.plot_init(on, col, col_equals, **kwargs)
         if df[plot_col].dtype == "bool":
             default_threshold = None
         else:
             default_threshold = "median"
         results = plot_kmf(
             df=df,
             condition_col=plot_col,
-            xlabel='Overall Survival' if how == "os" else 'Progression-Free Survival',
-            censor_col=self.deceased_col if how == "os" else self.progressed_or_deceased_col,
-            survival_col=self.os_col if how == "os" else self.pfs_col,
+            xlabel="Overall Survival" if how == "os" else "Progression-Free Survival",
+            censor_col="deceased" if how == "os" else "progressed_or_deceased",
+            survival_col=how,
             threshold=threshold if threshold is not None else default_threshold)
         print(results)
 
 def col_func(cohort, on, col, col_equals):
-    df = cohort.clinical_dataframe.copy()
+    df = cohort.as_dataframe()
     df[on] = df[col] == col_equals
     return on, df
+
+def verify_group_by(group_by):
+    if group_by not in ["patient", "paired_sample"]:
+        raise ValueError("Invalid group_by: %s" % group_by)
+
+def first_not_none(params, default):
```

Comment:
Yeah, sorry for the lack of comments. Wanted to get the PR out there but will update.